demo/src/lib/layout/Dropdown.svelte

```diff
@@ -0,0 +1,80 @@
+<script lang="ts">
+	import {writable} from '@amadeus-it-group/tansu';
+	import {createHasFocus} from '@agnos-ui/svelte-bootstrap/services/focustrack';
+	const open$ = writable(false);
+
+	// eslint-disable-next-line no-undef
+	type Item = $$Generic<{
+		id: string;
+		onclick?: () => void;
+		href?: string;
+		isSelected?: boolean;
+	}>;
+
+	const {hasFocus$, directive} = createHasFocus();
+	$: $open$ = $hasFocus$;
+
+	function giveFocus(el: HTMLAnchorElement | HTMLButtonElement, index: number) {
+		if (index === 0) {
+			el.focus();
+		}
+	}
+
+	const toggle = () => open$.update((val) => !val);
+
+	export let ariaLabel: string;
+	export let btnClass: string = '';
+	export let items: Item[];
+	export let mode: 'links' | 'buttons' = 'buttons';
+	export let placement: 'start' | 'end' = 'start';
+	export let dropdownClass: string = '';
+</script>
+
+<div class="dropdown {dropdownClass}">
+	<button
+		class="btn dropdown-toggle align-items-center d-flex {btnClass}"
+		aria-label={ariaLabel}
+		on:mousedown|preventDefault
+		on:click={toggle}
+		type="button"
+		aria-expanded={$open$}
+	>
+		<slot name="button" />
+	</button>
+	{#if $open$}
+		<div use:directive class="dropdown-menu dropdown-menu-{placement} position-absolute" class:show={$open$} data-bs-popper="absolute">
+			{#each items as item, index}
+				{#if mode === 'links'}
+					<a
+						use:giveFocus={index}
+						class="dropdown-item d-flex align-items-center"
+						href={item.href}
+						class:active={item.isSelected}
+						aria-current={item.isSelected ? 'page' : false}
+						on:click={toggle}
+					>
+						<slot name="item" {item} />
```

Comment:
```suggestion
						<slot name="item" {item} {index} />
```
Or should you remove the `if` with a generic `<svelte:element this={mode == "links" ? 'a' : 'button'} ...` ?